### PR TITLE
Wave2 480 bulk update for ownership

### PR DESCRIPTION
**Pull request by nindusaini**

### CODE REVIEW

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 1]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
     }
     return false;
 }
+
+function content_ownership_init() {
+    if (arg(0) == 'admin') {
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: And what does arg(0) represent...?

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 2]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+        if(isset($_POST['fpids'])){
+            $fpids = $_POST['fpids'];
+            $bulk_update_og_name = $_POST['update_og_name'];
+            //fpids are in a big long string, lets put them in an array for bulk processing
+            $bulk_update_fpids = explode(",", $fpids);
+            foreach($bulk_update_fpids as $key => $value){
+                //we might have couple of empty array values while converting from string so check those
+                if($value != ''){
+                    //See if ownership exist for this entity then just update it otherwise insert it
+                    $query = db_select('content_ownership', 'cown');
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: This code is duplicated four times?? Let's yank this out to a helper method, and call it as needed.

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 3]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+                            ->execute();
+                    }
+                }
+            }
+            if($query){
+                $form_state['storage']['success'] ='Ownership group updated successfully';
+                $form_state['rebuild'] = TRUE;
+            }
+        }
+    }elseif($form_args == 'panel_pages'){
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: Actually, if anything, just do a switch on the $form_args and figure out which value you need to process against.

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 4]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+        }
+    }
+}
+
+//this function display fpps present on selected group (e.g future-students or about pages)
+function getfpplist(){
+    $results = '<table>
+            <thead>
+            <tr>
+                <th>Title</th>
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: As a general rule, any text entered into form, like this, should be done with the t() method (eg. <th>' . t("Title") . '</td>) to ensure that content is translatable long term. 

That said, it's not a hard core rule for us (English-only) folks, so not an absolute need for change. Still, heads-up is always good to have.

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 5]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+                <th>Type</th>
+                <th>Page Title</th>
+            </tr>
+            </thead>
+            <tbody>';
+
+        $display_group = $_GET['group'];
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: Processing first, output second. Don't intermix them -- it makes it harder to trace.

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 6]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+                <thead>
+                <tr>
+                    <th>Title</th>
+                    <th>Type</th>
+                </tr>
+                </thead>
+                <tbody>';
+
+    //Get field name and node types that have FPPs as entity reference
+    $query = db_select('field_config', 'fc');
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: Again, process your data first, then present it all in one block.

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 7]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+    //'<input type="textbox" name="update_og_name" style="border:1px solid #000;" /> &nbsp;';
+    $results .= '<select name="update_og_name">';
+    foreach (taxonomy_get_tree($groups_taxonomy->vid) as $term){
+        $results .= '<option value="'.$term->tid.'">'.$term->name.'</option>';
+    }
+    return($results);
+
+}
+
+function getpageslist()
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: *coughCOMMENTScough*

File: sites/all/modules/ucalgary/content_ownership/content_ownership.module [hunk 8]
```diff
@@ -572,3 +572,869 @@ function _content_ownership_menu_access($string, $mlid, $menu_parent_id = NULL,
...
+function get_mini_panel_list(){
+    $results = '<table>
+            <thead>
+            <tr>
+                <th>Title</th>
+            </tr>
+            </thead>
+            <tbody>';
+
+    $display_group = $_GET['group'];
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/content_ownership.module]: As above: process first, present second.

File: sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-fpp-ownership.tpl.php [hunk 9]
```diff
@@ -0,0 +1,152 @@
+<?php if ($logged_in && $is_admin) { ?><!-- Include file: <?php echo str_replace('-', '=', __FILE__); ?> -->
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-fpp-ownership.tpl.php]: Is this file still being used?

File: sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-ownership-process.tpl.php [hunk 10]
```diff
@@ -0,0 +1,153 @@
+<?php if ($logged_in && $is_admin) { ?><!-- Include file: <?php echo str_replace('-', '=', __FILE__); ?> -->
```
Comment (gsowrey) [inline sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-ownership-process.tpl.php]: Is this file still being used?

Comment (nindusaini) [inline sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-ownership-process.tpl.php]: No, sorry forgot to delete, deleted it.

File: sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-fpp-ownership.tpl.php (hunk 9 above)
Comment (nindusaini) [inline sites/all/modules/ucalgary/content_ownership/templates/page--bulk-update-fpp-ownership.tpl.php]: No, sorry forgot to delete, deleted it as well.

**Comment by dpfitzsi** [comment]
Hi Navinder. Two files from this branch are now in conflict can they be resolved?

sites/all/modules/ucalgary/ucgy_program/ucgy_programs/templates/node--program.tpl.php
sites/all/modules/ucalgary/ucgy_program/ucgy_programs/templates/views-view-fields--program-requirements-v2--block-2.tpl.php

**Pull request merged**